src/nbl/video/CVKLogicalDevice.h

```diff
@@ -124,51 +327,472 @@ class CVKLogicalDevice final : public ILogicalDevice
+
+        VkImage vk_image;
+        if (vkCreateImage(m_vkdev, &vk_createInfo, nullptr, &vk_image) == VK_SUCCESS)
+        {
+            return core::make_smart_refctd_ptr<CVulkanImage>(core::smart_refctd_ptr<CVKLogicalDevice>(this),
+                std::move(params), vk_image);
+        }
+        else
+        {
+            return nullptr;
+        }
+    }
+
+    bool bindImageMemory(uint32_t bindInfoCount, const SBindImageMemoryInfo* pBindInfos) override
+    {
+        for (uint32_t i = 0u; i < bindInfoCount; ++i)
+        {
+            if (pBindInfos[i].image->getAPIType() != EAT_VULKAN)
+                continue;
+            VkImage vk_image = static_cast<const CVulkanImage*>(pBindInfos[i].image)->getInternalObject();
+
+            // if (pBindInfos[i].memory->getAPIType() != EAT_VULKAN)
+            //     continue;
+            VkDeviceMemory vk_deviceMemory = static_cast<const CVulkanMemoryAllocation*>(pBindInfos[i].memory)->getInternalObject();
+
+            if (vkBindImageMemory(m_vkdev, vk_image, vk_deviceMemory,
+                static_cast<VkDeviceSize>(pBindInfos[i].offset)) != VK_SUCCESS)
+            {
+                return false;
+            }
+        }
+
+        return true;
     }
             
     core::smart_refctd_ptr<IGPUImage> createGPUImageOnDedMem(IGPUImage::SCreationParams&& params, const IDriverMemoryBacked::SDriverMemoryRequirements& initialMreqs) override
     {
+        // Todo(achal)
+#if 0
+        core::smart_refctd_ptr<IGPUImage> gpuImage = createGPUImage(core::smart_refctd_ptr(params));
+        if (!gpuImage)
+            return nullptr;
+        
+        core::smart_refctd_ptr<video::IDriverMemoryAllocation> imageMemory =
+            allocateDeviceLocalMemory(gpuImage->getMemoryReqs());
+#endif
+
         return nullptr;
     }
 
     void updateDescriptorSets(uint32_t descriptorWriteCount, const IGPUDescriptorSet::SWriteDescriptorSet* pDescriptorWrites,
         uint32_t descriptorCopyCount, const IGPUDescriptorSet::SCopyDescriptorSet* pDescriptorCopies) override
     {
-        return;
+        constexpr uint32_t MAX_DESCRIPTOR_WRITE_COUNT = 25u;
+        constexpr uint32_t MAX_DESCRIPTOR_COPY_COUNT = 25u;
+        constexpr uint32_t MAX_DESCRIPTOR_ARRAY_COUNT = MAX_DESCRIPTOR_WRITE_COUNT;
+
+        // Todo(achal): This exceeds 16384 bytes on stack, move to heap
+
+        assert(descriptorWriteCount <= MAX_DESCRIPTOR_WRITE_COUNT);
+        VkWriteDescriptorSet vk_writeDescriptorSets[MAX_DESCRIPTOR_WRITE_COUNT];
+
+        uint32_t bufferInfoOffset = 0u;
+        VkDescriptorBufferInfo vk_bufferInfos[MAX_DESCRIPTOR_WRITE_COUNT * MAX_DESCRIPTOR_ARRAY_COUNT];
+
+        uint32_t imageInfoOffset = 0u;
+        VkDescriptorImageInfo vk_imageInfos[MAX_DESCRIPTOR_WRITE_COUNT * MAX_DESCRIPTOR_ARRAY_COUNT];
+
+        uint32_t bufferViewOffset = 0u;
+        VkBufferView vk_bufferViews[MAX_DESCRIPTOR_WRITE_COUNT * MAX_DESCRIPTOR_ARRAY_COUNT];
```

Comment:
really need that container from @Crisspl

Reply:
are there any updates on this?
CC: @Crisspl

Reply:
Related:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/160#discussion_r740689680